### ISSUE TITLE
Add a abbreviation list mechanism

### ISSUE THREAD

**Issue by denismaier** [citation-style-language/schema#321]
The question of abbreviations, I suggest should be handled by adopting lists of abbreviations for events and container titles. Zotero currently will automatically generate journal abbreviations from the MEDLINE list and has some hack-y methods to specify a different list. We should discuss whether a formally specified abbreviation list mechanism makes sense. I think it does.

_Originally posted by @bwiernik in https://github.com/citation-style-language/schema/pull/268#issuecomment-659661693_

I think we should absolutely add a mechanism for abbreviations.

**Comment by bdarcus** [comment]
Frank disagrees [here](https://discourse.citationstyles.org/t/design-principles-for-csl-json/1671/45).

**Comment by bwiernik** [comment]
This is sort of similar to the situation of multi-section bibliographies. We might include some hinting in the style (e.g., noting what abbreviations list is used), but it can probably stay at the calling application level for supplying the abbreviations.

**Comment by denismaier** [comment]
Yes. Currently there seems to exist an unofficial abbreviation list format supported by citeproc-js and pandoc-citeproc. Could be a good idea to formalize this so calling applications can supply them in this format.

**Comment by bdarcus** [comment]
We could define a simple JSON/YAML schema? What is that format they use?

**Comment by denismaier** [comment]
Here's how it looks:

```json
{ "default":  { 
"container-title": {
 "Lloyd's Law Reports": "Lloyd's Rep", 
"Estates Gazette": "EG", 
"Scots Law Times": "SLT" 
} } }
```

**Comment by bdarcus** [comment]
What's the significance of "default" there? What would be non-default

**Comment by bwiernik** [comment]
I believe that is indicating item type, with default being "all others".

**Comment by bwiernik** [comment]
Here is where Zotero loads the abbreviations file to pass to the processor: 
https://github.com/zotero/zotero/blob/91ca6d2ba6cbe6256a0adc97f3d3de05a9bcb833/chrome/content/zotero/xpcom/cite.js#L435

**Comment by denismaier** [comment]
In addition to the schema question I think this should address these scenarios:
a) an abbreviation file on a per document basis.
b) an abbreviation file on a per style basis.
c) an abbreviation file on a more general basis, say per discipline
d) a combination of a) b) and c)

**Comment by bwiernik** [comment]
The gist of what citeproc-js/pandoc-citeproc currently do:
  1. Take an abbreviations file
  2. Do abbreviation replacement based on that file when `form="short"`

The extent of support I think might be reasonable:
  1. Formalize this structure:
     1. The existence structure of abbreviations files
     1. The logic for replacement based on current behavior
  1. Provide an informational element in the style info indicating what abbreviations list a style uses.
    1. This wouldn't have to be used by the citation processor, but a calling application might use it to supply different abbreviations lists, e.g., MEDLINE for medical journals but Society of Biblical Literature's special list for that style.

**Comment by denismaier** [comment]
> Also, since we now have a `citation-key` this could be extended.

Here, I was thinking about something like this (using yaml for simplicity):

```yaml
citation-label:
  - id: item-1
    citation-label: XYZ
```

So, this would add `citation-label` for a specific item based on the `citation-key`.

**Comment by bwiernik** [comment]
I don't understand what that would be for? Why not just add `citation-label` to the item data? Can you give a real example?

**Comment by denismaier** [comment]
> Why not just add `citation-label` to the item data?

Because `citation-label` may be used only in this document. Say, your writing an essay on Freud's "Das Unbehagen in der Kultur", and want to use the label "UK" for this text. But you will use this label only in this document or maybe in a book Freud. But in another document you won't use this label.

**Comment by denismaier** [comment]
Just to add to this: This is mainly a problem for GUI apps where you'll build a huge career long database. That's much less of an issue if you're using pandoc and the like.

**Comment by bwiernik** [comment]
Hmm, three thoughts:
1. That is all about populating the `citation-label` field. That’s a fairly different case from automatically abbreviating journal or conference titles, which is about looking up form="short" content for a field (container-title) by the full length content. 
1. As you say, this is more an issue for GUI citation managers rather than a system like pandoc.

With that in mind, I’d suggest that citation-label is a problem for the calling application to solve. Either the user can enter it into the data, or the application can provide a way to fill the citation-label field for the item in the document (e.g., in Zotero, if a style calls citation-label, a text entry box could be added to the locator/suffix/prefix screen).

**Comment by denismaier** [comment]
Ok. Then this means we formalize the current structure.

> With that in mind, I’d suggest that citation-label is a problem for the calling application to solve. Either the user can enter it into the data, or the application can provide a way to fill the citation-label field for the item in the document (e.g., in Zotero, if a style calls citation-label, a text entry box could be added to the locator/suffix/prefix screen).

How do you think we should/could communicate this to calling application devs? Like: "CSL has a special variable `citation-label` that can be used in X and Y ways. For optimal UX please provide ways to do Z."
